pitest-entry/src/main/java/org/pitest/bytecode/analysis/MethodTree.java

```diff
@@ -74,9 +73,8 @@ public boolean isSynthetic() {
 
   private List<AbstractInsnNode> createInstructionList() {
     final List<AbstractInsnNode> list = new LinkedList<>();
-    final ListIterator<AbstractInsnNode> it = this.rawNode.instructions.iterator();
-    while (it.hasNext()) {
-        list.add(it.next());
+    for (AbstractInsnNode abstractInsnNode : this.rawNode.instructions) {
+      list.add(abstractInsnNode);
     }
     this.lazyInstructions = new ArrayList<>(list);
```

Comment:
I'm wondering why there is a double copy of the list. First when iterating over this.rawNode.instructions those instructions will be "copied"  into list. Afterwards list will be copied into new ArraList<>(list) . I don't think the second copy is necessary. We should use ArrayList from the beginning. Is the size/number of instructions known? Would make sense to initialize the ArrayList with the number of items before copying from instructions.

Reply:
removed double copy.